src/main/java/br/unicamp/cst/util/viewer/CodeletPanel.java

```diff
@@ -0,0 +1,356 @@
+                    String aprox;
+                    if (value.length == 1)
+                        aprox = String.format("%4.2f",value[0]);
+                    else if (value.length == 2)
+                        aprox = String.format("%4.2f,%4.2f",value[0],value[1]);
+                    else if (value.length == 3)
+                        aprox = String.format("%4.2f,%4.2f,%4.2f",value[0],value[1],value[2]);
+                    else if (value.length == 4)
+                        aprox = String.format("%4.2f,%4.2f,%4.2f,%4.2f",value[0],value[1],value[2],value[3]);
+                    else if (value.length == 5)
+                        aprox = String.format("%4.2f,%4.2f,%4.2f,%4.2f,%4.2f",value[0],value[1],value[2],value[3],value[4]);
+                    else if (value.length == 6)
+                        aprox = String.format("%4.2f,%4.2f,%4.2f,%4.2f,%4.2f,%4.2f",value[0],value[1],value[2],value[3],value[4],value[5]);
+                    else
+                        aprox = String.format("%4.2f,%4.2f,%4.2f,%4.2f,%4.2f,,%4.2f...",value[0],value[1],value[2],value[3],value[4],value[5]);
+                    memValue = aprox+isize;
+                }
+                else memValue = o.toString()+isize;
+                return(memValue);
+            }   
+        }
+        return(memValue);
+    }
+    
+    public String getMemoryValue2(String nodeName) {
+        String memValue = "";
+        List<Memory> mm = mind.getRawMemory().getAllOfType(nodeName);
+        if (mm.size() > 0){
+            Object obj = mm.get(0).getI();
+            double value=0;
+            if (obj != null) {
+                if (obj instanceof Double) {
+                    value = (double)obj;
+                    memValue = toString(value);
+                }
+                else memValue = obj.toString();
+            }
+            else memValue = "null";
+        }
+        else {
+            // Test if the nodeName is an MO within an MC
+            String[] mc = nodeName.split("\\(");
+            if (mc.length > 1) {
+                String[] mc2 = mc[1].split("\\)");
+                int subNode = -1;
+                try {
+                    subNode = Integer.parseInt(mc2[0]);
+                } catch(NumberFormatException e) {
+                    e.printStackTrace();
+                }
+                mm = mind.getRawMemory().getAllOfType(mc[0]);
+                if (mm.size() > 0) {
+                    Memory m = mm.get(0);
+                    MemoryContainer mmc;
+                    if (m.getClass().getCanonicalName().equalsIgnoreCase("br.unicamp.cst.core.entities.MemoryContainer")) {
+                        mmc = (MemoryContainer) m;
+                        double value = (double) mmc.getI(subNode);
+                        memValue = toString(value);
+                    }    
+                }
+            } 
+            
+        }
+        return(memValue);
+    }
+    
+    public void updateTree(Mind m) {
```

Comment:
The parameter `Mind m` is never used inside this method. Why not remove it?

Reply:
This was due to legacy ... it was required in the old MindViewer code ... but you are right .. it is not necessary anymore. I will remove it.